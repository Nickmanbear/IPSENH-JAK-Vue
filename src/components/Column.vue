<template>
  <div id="column">
    <h2>Title</h2>
    <div id="cards">
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
      <Card />
    </div>
    <p>+ Add card</p>
  </div>
</template>

<script>
// @ is an alias to /src
import Card from '@/components/CardPreview.vue';

export default {
  name: 'Home',
  components: {
    Card,
  },
};
</script>

<style lang="scss">
  #column {
    background-color: #eee;
    border: 1px solid #eee;
    border-radius: 3px;
    padding: 5px 10px;
    margin: 5px;
    display: inline-block;
    width: 240px;
    max-height: 80vh;
  }

  h2 {
    margin: 5px 0;
  }

  #cards {
    height: 70vh;
    overflow: scroll;
    -ms-overflow-style: none;
  }

  #cards::-webkit-scrollbar {
    display: none;
  }
</style>
