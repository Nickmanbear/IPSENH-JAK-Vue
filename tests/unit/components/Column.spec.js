import { shallowMount } from '@vue/test-utils';
import Column from '@/components/Column.vue';

jest.mock('axios', () => ({
  create: () => ({
    post: () => Promise.resolve({
      data: {
        id: 1,
        column: { id: 1, board: { id: 1, users: [{ id: 1 }] } },
        name: 'new card',
        description: '',
      },
    }),
    get: () => Promise.resolve({ data: [] }),
    delete: () => Promise.resolve(),
  }),
}));

describe('Column', () => {
  let cmp;

  beforeEach(() => {
    cmp = shallowMount(Column, {
      propsData: {
        column: {
          id: 1,
          board: { id: 1 },
          name: 'test column',
        },
      },
    });
    window.confirm = jest.fn(() => true);
  });

  it('should delete itself', async () => {
    await cmp.vm.deleteColumn();
    expect(cmp.emitted().deleted).toBeTruthy();
  });

  it('should stop editing the name', () => {
    cmp.vm.editingName = true;
    cmp.vm.saveName();
    expect(cmp.vm.editingName).toBeFalsy();
  });

  it('should create a new card', async () => {
    cmp.vm.newCardName = 'new card';
    await cmp.vm.createCard();
    expect(cmp.vm.cards).toEqual([{
      id: 1,
      column: { id: 1, board: { id: 1, users: [{ id: 1 }] } },
      name: 'new card',
      description: '',
    }]);
  });

  it('should reload its cards', async () => {
    cmp.vm.cards = ['test', 'test2', 'test3'];
    await cmp.vm.removeCard('test2');
    expect(cmp.vm.cards).toEqual(['test', 'test3']);
  });

  it('should react to a moved card', () => {
    cmp.vm.moveCard({ item: { _underlying_vm_: { column: { id: 1} } } });
  });

  it('equals name to "test column"', () => {
    expect(cmp.find('h2').text()).toEqual('test column');
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
