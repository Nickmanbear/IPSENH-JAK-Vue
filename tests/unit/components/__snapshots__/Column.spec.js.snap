// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column has the expected html structure 1`] = `
<div
  id="column"
>
  <h2>
    test column
  </h2>
   
  <!---->
   
  <button
    id="delete"
  >
    Ã—
  </button>
   
  <hr />
   
  <div
    id="cards"
  >
    <draggable-stub
      clone="function(t){return t}"
      element="div"
      group="cards"
      value=""
    />
  </div>
   
  <button
    id="create"
  >
    + Add card
  </button>
   
  <!---->
</div>
`;
